edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Http/MethodResult.cs

```diff
@@ -13,23 +14,31 @@ public MethodResult(int status, JRaw payload)
         }
 
         [JsonProperty("status")]
-        public int Status { get; }
+        public virtual int Status { get; }
 
         [JsonProperty("payload")]
-        public JRaw Payload { get; }
+        public virtual JRaw Payload { get; }
     }
 
     public class MethodErrorResult : MethodResult
```

Comment:
would you explain why we need to hide Status and Payload in json for Error case? What is the difference between Status (int) and StatusCode (HttpStatusCode)?  should we just implement MethodErrorResult without inheriting from MethodResult?

Reply:
is it because MethodResult is used for return type?  if so, should we have a base class MethodResult without any property; then we have 2 classes inherited from it - MethodErrorResult and MethodSuccessResult?  What do you think?

Reply:
status vs status code: in normal case, the http response has its usual http status code (e.g. 200-ok). the returned http content, which is a json structure, also has a status property, which needs to be carried.
The suggestion about the inheritance (create an empty base class and two derived for the two case) makes sense.